src/applications/model/gaming-traffic-model.h

```diff
@@ -0,0 +1,86 @@
+/* -*- Mode:C++; c-file-style:"gnu"; indent-tabs-mode:nil; -*- */
+/*
+ * Copyright (c) 2020, University of Padova, Department of Information
+ * Engineering, SIGNET Lab.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation;
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+ *
+ * Authors: Salman Mohebi <s.mohebi22@gmail.com>
+ *
+ */
+
+#ifndef GAMING_TRAFFIC_MODEL_H
+#define GAMING_TRAFFIC_MODEL_H
+
+#include "ns3/type-id.h"
+#include "ns3/object.h"
+#include "ns3/attribute-helper.h"
+#include "ns3/random-variable-stream.h"
+#include <stdint.h>
+
+namespace ns3 {
+/**
+ * Random number generator based on different distributions
+ * with different probabilities
+ */
+  class GamingTrafficModel : public RandomVariableStream
+  {
+  public:
+    /**
+   * \brief Register this type.
+   * \return The object TypeId.
+   */
+    static TypeId GetTypeId (void);
+
+    /**
+    * \brief Creates GamingTrafficModel object with the default Random variables
+    */
+    GamingTrafficModel ();
+
+     /**
+     * \brief Returns a random double from defined distributions based on their probabilities
+     * \return A floating point random value.
+    */
+    virtual double GetValue(void);
+
+    /**
+     * \brief Returns a random unsigned integer from defined distributions based on their probabilities
+     * \return A random unsigned integer value.
+    */
+    virtual uint32_t GetInteger (void);
+
+    /**
+     * Set the probability of RandomVariables
+     */
+    void SetProb (std::vector < double >);
+
+    /**
+     * \brief Set the RandomVariables vector
+     */
+    void SetRngs (std::vector <Ptr<RandomVariableStream> >);
+
+  private:
+    /**
+     * Check if the probability function is valid
+     * @return True if the sum of m_prob is equal to one and False otherwise
+     */
+    bool IsValid (void);
+
+    std::vector <Ptr<RandomVariableStream> > m_rng;    //!< Vector of RandomVariables
+    std::vector < double > m_prob;  //!< Vector of Probabilities for RandomVariables
```

Comment:
I suggest you to use better variable names. For example `m_probs` to indicate that it's actually a vector, not a single number, and `m_rvs` for "Random Variables". Rng, in fact, is not the name you're looking for since it doesn't refer to a random variable but rather to the whole mechanism involved in generating a stream of random variables